portability-core/src/main/java/org/dataportabilityproject/job/JobUtils.java

```diff
@@ -42,32 +43,38 @@ public static String encodeId(PortabilityJob job) {
     return BaseEncoding.base64Url().encode(job.id().getBytes(Charsets.UTF_8));
   }
 
-  /* Returns the initial auth data for export or import determined by the {@code isExport} param. */
-  public static AuthData getInitialAuthData(PortabilityJob job, boolean isExport) {
-    return isExport ? job.exportInitialAuthData() : job.importInitialAuthData();
+  /* Returns the initial auth data for export or import determined by the {@code serviceMode} param. */
+  public static AuthData getInitialAuthData(PortabilityJob job, ServiceMode serviceMode) {
+    return serviceMode == ServiceMode.EXPORT? job.exportInitialAuthData() : job.importInitialAuthData();
   }
 
   /* Sets the service in the correct field of the PortabilityJob */
   public static PortabilityJob setAuthData(PortabilityJob job, AuthData authData,
-      boolean isExportService) {
+      ServiceMode serviceMode) {
     PortabilityJob.Builder updatedJob = job.toBuilder();
-    if (isExportService) {
-      updatedJob.setExportAuthData(authData);
-    } else {
-      updatedJob.setImportAuthData(authData);
+    switch (serviceMode) {
+      case EXPORT:
+        updatedJob.setExportAuthData(authData);
+        break;
+      case IMPORT:
+        updatedJob.setImportAuthData(authData);
+        break;
     }
     return updatedJob.build();
   }
 
   /* Sets the service in the correct field of the PortabilityJob */
   public static PortabilityJob setInitialAuthData(PortabilityJob job, AuthData initialAuthData,
-      boolean isExport) {
-    logger.debug("Setting initialAuthData: {}, isExport:  {}", initialAuthData, isExport);
+      ServiceMode serviceMode) {
+    logger.debug("Setting initialAuthData: {}, serviceMode: {}", initialAuthData, serviceMode);
     PortabilityJob.Builder updatedJob = job.toBuilder();
-    if (isExport) {
-      updatedJob.setExportInitialAuthData(initialAuthData);
-    } else {
-      updatedJob.setImportInitialAuthData(initialAuthData);
+    switch (serviceMode) {
+      case EXPORT:
+        updatedJob.setExportInitialAuthData(initialAuthData);
+        break;
+      case IMPORT:
+        updatedJob.setImportInitialAuthData(initialAuthData);
+        break;
```

Comment:
Add case default that throws exception "Unsupported ServiceMode"

Reply:
Done.